core/launch_configuration.go

```diff
@@ -85,33 +86,24 @@ func (lc *launchConfiguration) convertLaunchConfigurationToSpotSpecification(
 		}
 	}
 
+	secGroupIDs, err := getSecurityGroupIDs(conn, lc.SecurityGroups)
+	if err != nil {
+		return nil, err
```

Comment:
I think this error condition could indicate we're running on EC2 Classic, where we can go ahead instead of returning here, and use the names as they are instead of using IDs.

I'd try to test this on one of our EC2-Classic environments to see what it actually happens.

Reply:
IIRC if we are on EC2-Classic, the securitygroups will be given as names. I would expect those to just be converted successfully to IDs here (if the API call is succesfull) and no error to be returned.